pkg/controller/worker/machines.go

```diff
@@ -138,6 +138,10 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			"secret": map[string]interface{}{
 				"cloudConfig": string(pool.UserData),
 			},
+			"credentialsSecretRef": map[string]interface{}{
+				"name":      "cloudprovider",
```

Comment:
You could use `v1beta1constants.SecretNameCloudProvider`

Reply:
I like that better.